### ISSUE TITLE
Write coagulation strategy system feature documentation

### ISSUE THREAD

**Issue by Gorkowski** [uncscode/particula#860]
## Description

Create comprehensive feature documentation for the coagulation strategy system following the format established in `docs/Features/wall_loss_strategy_system.md`. This documentation will serve as the primary user-facing guide for understanding and using particula's coagulation capabilities.

## Context

The coagulation module (`particula/dynamics/coagulation/`) is a mature, feature-rich subsystem that currently lacks user-facing feature documentation. The wall loss strategy system has an excellent feature doc that serves as a template. Users need similar documentation for coagulation to understand:

- Available coagulation strategies (Brownian, Charged, Turbulent Shear, Turbulent DNS, Sedimentation)
- How to use builders and factories
- Distribution type support (discrete, continuous_pdf, particle_resolved)
- How coagulation integrates with other dynamics (condensation, wall loss)
- The `Coagulation` runnable for pipeline composition

**Value:**
- Enables users to quickly understand coagulation capabilities
- Provides clear examples for common use cases
- Documents the strategy/builder/factory pattern used consistently across dynamics
- Serves as a reference for new feature development

## Scope

**Estimated Lines:** ~350 lines (matching wall_loss_strategy_system.md length)
**Complexity:** Medium (documentation writing, no code changes)

**Files to Create:**
- `docs/Features/coagulation_strategy_system.md` (~350 lines)

**Reference Files to Read:**
- `docs/Features/wall_loss_strategy_system.md` - Template to follow
- `particula/dynamics/coagulation/coagulation_strategy/coagulation_strategy_abc.py` - Base class
- `particula/dynamics/coagulation/coagulation_strategy/brownian_coagulation_strategy.py` - Brownian impl
- `particula/dynamics/coagulation/coagulation_strategy/charged_coagulation_strategy.py` - Charged impl
- `particula/dynamics/coagulation/coagulation_strategy/turbulent_shear_coagulation_strategy.py` - Turbulent shear
- `particula/dynamics/coagulation/coagulation_strategy/turbulent_dns_coagulation_strategy.py` - Turbulent DNS
- `particula/dynamics/coagulation/coagulation_strategy/sedimentation_coagulation_strategy.py` - Sedimentation
- `particula/dynamics/coagulation/coagulation_strategy/combine_coagulation_strategy.py` - Combined strategies
- `particula/dynamics/coagulation/coagulation_builder/` - All builders
- `particula/dynamics/coagulation/coagulation_factories.py` - Factory
- `particula/dynamics/particle_process.py` - Coagulation runnable
- `particula/dynamics/__init__.py` - Public API exports

### Documentation Structure (following wall_loss template)
- [ ] **Overview** section explaining coagulation strategy system purpose
- [ ] **Key Benefits** section with 4-5 bullet points
- [ ] **Who It's For** section targeting users, developers, researchers
- [ ] **Capabilities** section covering:
  - [ ] Unified coagulation API in `particula.dynamics`
  - [ ] Runnable entry point: `Coagulation`
  - [ ] Each strategy type (Brownian, Charged, Turbulent Shear, DNS, Sedimentation)
  - [ ] Combined coagulation strategies
  - [ ] Builder and factory workflow
  - [ ] Support for multiple distribution types
- [ ] **Getting Started** section with quick start example
- [ ] **Prerequisites** section
- [ ] **Typical Workflows** section (3 workflows)
- [ ] **Use Cases** section (2-3 use cases)
- [ ] **Configuration** table with options/defaults
- [ ] **Best Practices** section (4-5 practices)
- [ ] **Limitations** section
- [ ] **Related Documentation** links
- [ ] **FAQ** section (2-3 questions)
- [ ] **See Also** links

### Code Examples
- [ ] Quick start example with BrownianCoagulationStrategy
- [ ] Example using ChargedCoagulationStrategy
- [ ] Example using CombineCoagulationStrategy
- [ ] Example with Coagulation runnable and pipeline composition
- [ ] Builder/factory workflow examples
- [ ] Particle-resolved coagulation example

### Quality
- [ ] All code examples are syntactically correct and runnable
- [ ] Cross-references to existing docs are valid markdown links
- [ ] Follows Google-style formatting for code blocks
- [ ] No broken internal links
- [ ] Consistent with wall_loss_strategy_system.md formatting

### Key Classes to Document

```python
# Strategies (exported via particula.dynamics)
par.dynamics.CoagulationStrategyABC        # Abstract base
par.dynamics.BrownianCoagulationStrategy   # Brownian diffusion
par.dynamics.ChargedCoagulationStrategy    # Coulomb-enhanced
par.dynamics.TurbulentShearCoagulationStrategy  # Turbulent shear
par.dynamics.TurbulentDNSCoagulationStrategy    # Turbulent DNS
par.dynamics.SedimentationCoagulationStrategy   # Gravitational settling
par.dynamics.CombineCoagulationStrategy    # Multiple mechanisms

# Builders
par.dynamics.BrownianCoagulationBuilder
par.dynamics.ChargedCoagulationBuilder
par.dynamics.TurbulentShearCoagulationBuilder
par.dynamics.TurbulentDNSCoagulationBuilder
par.dynamics.SedimentationCoagulationBuilder
par.dynamics.CombineCoagulationStrategyBuilder

# Factory and Runnable
par.dynamics.CoagulationFactory
par.dynamics.Coagulation  # RunnableABC implementation
```

### Strategy Interface Methods

All coagulation strategies implement:
- `kernel(particle, temperature, pressure)` → coagulation kernel [m³/s]
- `loss_rate(particle, kernel)` → loss rate [kg/s]
- `gain_rate(particle, kernel)` → gain rate [kg/s]
- `net_rate(particle, temperature, pressure)` → net rate [kg/s]
- `step(particle, temperature, pressure, time_step)` → updated particle

### Distribution Types Supported
- `"discrete"` - radius-binned distributions
- `"continuous_pdf"` - continuous probability-density representations
- `"particle_resolved"` - individual particle ensembles

## Example Usage to Document

```python
import particula as par

# Quick start with Brownian coagulation
strategy = par.dynamics.BrownianCoagulationStrategy(
    distribution_type="discrete"
)

particle = par.particles.PresetParticleRadiusBuilder().build()
T, P = 298.15, 101325.0

# Compute kernel and rates
kernel = strategy.kernel(particle, temperature=T, pressure=P)
net_rate = strategy.net_rate(particle, temperature=T, pressure=P)

# Advance one timestep
particle = strategy.step(particle, temperature=T, pressure=P, time_step=1.0)

# Using the Coagulation runnable
coagulation = par.dynamics.Coagulation(
    coagulation_strategy=strategy
)
aerosol = coagulation.execute(aerosol, time_step=60.0)

# Pipeline composition
wall_loss = par.dynamics.WallLoss(wall_loss_strategy=...)
combined = coagulation | wall_loss
aerosol = combined.execute(aerosol, time_step=60.0)
```

## References

**Template:**
- `docs/Features/wall_loss_strategy_system.md` - Follow this structure

**Source Code:**
- `particula/dynamics/coagulation/` - All coagulation modules
- `particula/dynamics/__init__.py` - Public API

**Coding Standards:**
- `adw-docs/docstring_guide.md` - Documentation style
- `adw-docs/code_style.md` - Code example formatting

**Comment by Gorkowski** [comment]
[ADW-AGENTS] # 🤖 ADW Workflow Status

**Description:** Document workflow: plan → document → ship

**ADW ID:** `e68b106d`
**Current Phase:** Executing: Generate Documentation
**Started:** 2026-01-16T18:39:18.064831Z
**Last Update:** 2026-01-16T18:39:29.674029Z

## Current Status

⚙️ **Generate Documentation**
**Elapsed Time:** 11s

## Workflow Plan

✅ Generate Multi-Review Implementation Plan _(at 2026-01-16T18:39:28.158073Z)_
⚙️ Generate Documentation
⏳ Ship Implementation

<details open>
<summary>📋 Recent Activity</summary>

```
[2026-01-16T18:39:18.110097Z] Workflow document started with 3 steps
[2026-01-16T18:39:20.001511Z] Starting step 1/3: Generate Multi-Review Implementation Plan
[2026-01-16T18:39:28.158089Z] Completed step 1/3: Generate Multi-Review Implementation Plan
[2026-01-16T18:39:29.671985Z] Starting step 2/3: Generate Documentation
```

</details>

---
_This comment is automatically updated as the workflow progresses_

**Comment by Gorkowski** [comment]
[ADW-AGENTS] # 🤖 ADW Workflow Status

**Description:** Document workflow: plan → document → ship

**ADW ID:** `16b158a1`
**Current Phase:** Executing: Generate Multi-Review Implementation Plan
**Started:** 2026-01-16T19:02:28.240253Z
**Last Update:** 2026-01-16T19:02:30.151659Z

## Current Status

⚙️ **Generate Multi-Review Implementation Plan**
**Elapsed Time:** 1s

## Workflow Plan

⚙️ Generate Multi-Review Implementation Plan
⏳ Generate Documentation
⏳ Ship Implementation

<details open>
<summary>📋 Recent Activity</summary>

```
[2026-01-16T19:02:28.284198Z] Workflow document started with 3 steps
[2026-01-16T19:02:30.149354Z] Starting step 1/3: Generate Multi-Review Implementation Plan
```

</details>

---
_This comment is automatically updated as the workflow progresses_
